Failed to build Agent modules with JDK 20

## Build

```bash
JAVA_HOME=$JAVA20_HOME ./mvnw clean install -T1C -f agent/pom.xml -nsu
```


```
Caused by: java.lang.IllegalArgumentException: Java 20 (64) is not supported by the current version of Byte Buddy which officially supports Java 19 (63) - update Byte Buddy or set net.bytebuddy.experimental as a VM property
        at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:96)
        at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3948)
        at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2213)
        at net.bytebuddy.dynamic.scaffold.inline.RedefinitionDynamicTypeBuilder.make(RedefinitionDynamicTypeBuilder.java:224)
        at net.bytebuddy.dynamic.scaffold.inline.AbstractInliningDynamicTypeBuilder.make(AbstractInliningDynamicTypeBuilder.java:123)
        at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3668)
        at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.transform(InlineBytecodeGenerator.java:401)
        at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:244)
        at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
        at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:541)
        at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
        at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:169)
        at org.mockito.internal.creation.bytebuddy.InlineBytecodeGenerator.triggerRetransformation(InlineBytecodeGenerator.java:280)
        ... 93 more
```


byte-buddy supports Java 20 since 1.12.18. But after bumped, compile error occurred.

```
diff --git a/agent/pom.xml b/agent/pom.xml
index a8d8f946ebf..a89138394f4 100644
--- a/agent/pom.xml
+++ b/agent/pom.xml
@@ -35,7 +35,7 @@
     </modules>
     
     <properties>
-        <bytebuddy.version>1.12.12</bytebuddy.version>
+        <bytebuddy.version>1.12.18</bytebuddy.version>
     </properties>
     
     <dependencyManagement>
```

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project shardingsphere-agent-core: Compilation failure: Compilation failure: 
[ERROR] /home/wuweijie/projects/shardingsphere/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java:[49,14] org.apache.shardingsphere.agent.core.builder.AgentTransformer is not abstract and does not override abstract method transform(net.bytebuddy.dynamic.DynamicType.Builder<?>,net.bytebuddy.description.type.TypeDescription,java.lang.ClassLoader,net.bytebuddy.utility.JavaModule,java.security.ProtectionDomain) in net.bytebuddy.agent.builder.AgentBuilder.Transformer
[ERROR] /home/wuweijie/projects/shardingsphere/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java:[64,5] method does not override or implement a method from a supertype
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -87,7 +87,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java-version: [ 11, 19 ]
+        java-version: [ 11, 20 ]
     steps:
       - uses: actions/checkout@v3
       - name: Cache Maven Repos

--- a/.github/workflows/nightly-ci.yml
+++ b/.github/workflows/nightly-ci.yml
@@ -37,7 +37,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        java-version: [ 11, 17, 19 ]
+        java-version: [ 11, 17, 20 ]
     steps:
       - name: Support long paths in Windows
         if: matrix.os == 'windows-latest'

--- a/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java
+++ b/agent/core/src/main/java/org/apache/shardingsphere/agent/core/builder/AgentTransformer.java
@@ -37,6 +37,7 @@ import org.apache.shardingsphere.agent.core.plugin.PluginLifecycleServiceManager
 import org.apache.shardingsphere.agent.core.plugin.classloader.AgentPluginClassLoader;
 import org.apache.shardingsphere.agent.core.plugin.classloader.ClassLoaderContext;
 
+import java.security.ProtectionDomain;
 import java.util.Collection;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
@@ -62,7 +63,7 @@ public final class AgentTransformer implements Transformer {
     
     @SuppressWarnings("NullableProblems")
     @Override
-    public Builder<?> transform(final Builder<?> builder, final TypeDescription typeDescription, final ClassLoader classLoader, final JavaModule module) {
+    public Builder<?> transform(final Builder<?> builder, final TypeDescription typeDescription, final ClassLoader classLoader, final JavaModule module, final ProtectionDomain protectionDomain) {
         if (!advisorConfigs.containsKey(typeDescription.getTypeName())) {
             return builder;
         }

--- a/agent/pom.xml
+++ b/agent/pom.xml
@@ -35,7 +35,7 @@
     </modules>
     
     <properties>
-        <bytebuddy.version>1.12.12</bytebuddy.version>
+        <bytebuddy.version>1.14.4</bytebuddy.version>
     </properties>
     
     <dependencyManagement>

